packages/form-js-viewer/src/core/ConditionChecker.js

```diff
@@ -0,0 +1,79 @@
+import { unaryTest } from 'feelin';
+import { isString } from 'min-dash';
+
+export class ConditionChecker {
+  constructor(formFieldRegistry) {
+    this._formFieldRegistry = formFieldRegistry;
+  }
+
+  /**
+   * For given data, remove properties for which conditions are not met.
+   *
+   * @param {Object<string, any>} properties
+   * @param {Object<string, any>} data
+   */
+  applyConditions(properties, data = {}) {
+    const conditions = this._getConditions();
+
+    const newProperties = { ...properties };
+    for (const { key, condition } of conditions) {
+      const shouldRemove = !this.check(condition, data);
+
+      if (shouldRemove) {
+        delete newProperties[ key ];
+      }
+    }
+
+    return newProperties;
+  }
+
+  /**
+   * Check if given condition is met.
+   *
+   * @param {string} condition
+   * @param {import('../types').Data} [data]
+   *
+   * @returns {boolean}
+   */
+  check(condition, data = {}) {
+    if (!condition) {
+      return true;
+    }
+
+    if (!isString(condition) || !condition.startsWith('=')) {
+      return false;
+    }
+
+    try {
+      const result = unaryTest(condition.slice(1), data);
```

Comment:
I guess at this point we don't want to enable context sensitive parsing, right? Our inputs don't contain spaces and other special chars.

Reply:
Not sure what you mean. I need to pass data in order to correctly evaluate the condition, right?